core/sql/parser/sqlparser.y

```diff
@@ -6579,6 +6581,9 @@ table_reference : table_name_and_hint
               | rel_subquery_and_as_clause
                 { $$ = $1; }
 
+              | with_clause_list
```

Comment:
I think this is incorrect. Making "with_clause_list" an alternative for table_reference would allow things like: "select * from with c1 as (select * from t)".